src/ConnectedMode/UI/Credentials/CredentialsViewModel.cs

```diff
@@ -0,0 +1,108 @@
+{
+    private string token;
+    private string selectedAuthenticationType = UiResources.Token;
+    private string username;
+    private string password;
+
+    public Connection Connection { get; } = connection;
+    public ObservableCollection<string> AuthenticationType { get; } = [UiResources.Token, UiResources.Credentials];
+
+    public string SelectedAuthenticationType
+    {
+        get => selectedAuthenticationType;
+        set
+        {
+            selectedAuthenticationType = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsTokenAuthentication));
+            RaisePropertyChanged(nameof(IsCredentialsAuthentication));
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
+            RaisePropertyChanged(nameof(ShouldTokenBeFilled));
+            RaisePropertyChanged(nameof(ShouldUsernameBeFilled));
+            RaisePropertyChanged(nameof(ShouldPasswordBeFilled));
+        }
+    }
+
+    public string Token
+    {
+        get => token;
+        set
+        {
+            token = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
+            RaisePropertyChanged(nameof(ShouldTokenBeFilled));
+        }
+    }
+
+    public string Username
+    {
+        get => username;
+        set
+        {
+            username = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
+            RaisePropertyChanged(nameof(ShouldUsernameBeFilled));
+        }
+    }
+
+    public string Password
+    {
+        get => password;
+        set
+        {
+            password = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
+            RaisePropertyChanged(nameof(ShouldPasswordBeFilled));
+        }
+    }
+
+    public bool IsTokenAuthentication => SelectedAuthenticationType == UiResources.Token;
+    public bool IsCredentialsAuthentication => SelectedAuthenticationType == UiResources.Credentials;
+    public bool ShouldTokenBeFilled => IsTokenAuthentication && !IsTokenProvided;
+    public bool ShouldUsernameBeFilled => IsCredentialsAuthentication && !IsUsernameProvided;
+    public bool ShouldPasswordBeFilled => IsCredentialsAuthentication && !IsPasswordProvided;
+    public bool IsConfirmationBtnEnabled => (IsTokenAuthentication && IsTokenProvided) || (IsCredentialsAuthentication && AreCredentialsProvided);
+
+    private bool IsTokenProvided => IsValueFilled(Token);
+    private bool AreCredentialsProvided => IsPasswordProvided && IsUsernameProvided;
+    private bool IsUsernameProvided => IsValueFilled(Username);
+    private bool IsPasswordProvided => IsValueFilled(Password);
+    public string GenerateTokenUrl => Connection.serverType == ServerType.SonarCloud ? SonarCloudGenerateTokenUrl : Path.Combine(Connection.id, SonarQubeGenerateTokenUrl);
```

Comment:
Could we disable this easily **if** we cut the branch during the next sprint?

Reply:
Ah wait this is just a hyperlink to the security page? Then I think we might need to change the name, because token generation is a bit more than that and involves communication between the ide (through SLCore) with the server to automatically exchange the token

Reply:
Maybe I can name it "AccountSecurityUrl" as it is a link to that page, where the generate token button in the server is present.